serverless/serverless
# Fix for missing artifact of layers with prebuilt artifacts

Problem: When a layer defines an artifact using 
```
layers:
   <LayerName>:
       package:
            artifact: <Artifact File|i.e: xyz.zip>
```

These artifacts dont get copied to the < WorkingDir>/.serverless directory as a result of the package or deploy command line executions. This causes the package to be incomplete or for the deployment to fail as the file cannot be found to upload it to S3.




## What did you implement:

Closes #XXXXX



## How did you implement it:

When a layer is being packaged, if the layer contains an artifact file, its copied from the path provided on layers.< layername>.package.artifact property,  to < WorkingDir>/.serverless/< BaseName>.  The property layers.< LayerName>.package.artifact gets updated to < BaseName>, where basename is the basename of the original path provided for the artifact.



## How can we verify it:

Add a layer with an artifact before the patch at the file will not be copied, after the patch the file will be as expected in the <WorkingDir>/.serverless directory.

Use following.Yaml snippet

```
layers:
   testlayer:
       package:
            artifact: ./artifacts/test.zip
```

Make sure <WorkingDir>/artifacts.test.zip exists.


## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
